server/index.js

```diff
@@ -0,0 +1,155 @@
+const express = require('express');
+const app = express();
+
+const PORT = process.env.PORT || 8000;
+
+// const knex = require('./middleware/knex');
+const APP_MODE = 'development';
+const Knex = require('knex');
+const dbConfig = require('./knexfile');
+const knex = Knex(dbConfig[APP_MODE]);
+
+const bcrypt = require('bcrypt-as-promised');
+// const cookieSession = require('cookie-session');
+
+const bodyParser = require('body-parser');
+app.use(bodyParser.json());
+app.use(bodyParser.urlencoded({extended: false}));
+
+const request = require('request');
+const cheerio = require('cheerio');
+
+// app.use(cookieSession({
+//   name: 'auth_app_user',
+//   keys: ['socure1', 'socure2']
+// }));
+
+//CORS for Client
+app.options('/*', (req, res, next) => {
```

Comment:
Would probably change this to be:

```js
const OPTIONS = 'OPTIONS';

// Pull out CORS headers into a constant.
const CORS_HEADERS = 
  Object.freeze({
      'Access-Control-Allow-Origin': '*',
      'Access-Control-Allow-Headers': 'Content-Type, authorization',
      'Access-Control-Allow-Credentials': 'true',
      'Access-Control-Allow-Methods':  'GET, POST, OPTIONS, PUT, PATCH',
   });

app.use('/*', (req, res, next) => {
  res.set(CORS_HEADERS);

  if (req.method === OPTIONS) {
    // Handling for preflight requests
     res.sendStatus(200); 
  } else  {
     next();
  }
});
```

You could also break it into a middleware and `options` handler.

```js
const ANY_PATH = '/*';

const CORS_HEADERS = 
  Object.freeze({
      'Access-Control-Allow-Origin': '*',
      'Access-Control-Allow-Headers': 'Content-Type, authorization',
      'Access-Control-Allow-Credentials': 'true',
      'Access-Control-Allow-Methods':  'GET, POST, OPTIONS, PUT, PATCH',
   });

app.use(ANY_PATH, (req, res, next) => {
     res.set(CORS_HEADERS);
     next();
});

app.options(ANY_PATH, (req, res) => { res.sendStatus(200); })
```


Good reading on the CORS handling: [MDN Access Control](https://developer.mozilla.org/en-US/docs/Web/HTTP/Access_control_CORS)